File: commonlibrary/src/main/java/com/jqk/commonlibrary/util/ScreenUtil.java
package com.jqk.commonlibrary.util;

import android.content.Context;
import android.util.DisplayMetrics;

public class ScreenUtil {
    /**
     * 获取屏幕密度
     * 单位：Float
     */
    public static float getDensity(Context context) {
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        float density = dm.density;
        return density;
    }

    /**
     * 获取屏幕宽度
     * 单位：Int
     */
    public static int getScreenWidth(Context context) {
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        int screenWidth = dm.widthPixels;
        return screenWidth;
    }

    /**
     * 获取屏幕高度
     * 单位：Int
     */
    public static int getScreenHeight(Context context) {
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        int screenHeight = dm.heightPixels;
        return screenHeight;
    }

    /**
     * 获取状态栏高度
     */
    public static int getStatusBarHeight(Context context) {
        int result = 0;
        int resourceId = context.getResources().getIdentifier("status_bar_height", "dimen", "android");
        if (resourceId > 0) {
            result = context.getResources().getDimensionPixelSize(resourceId);
        }
        return result;
    }
}

File: app/src/main/java/com/jqk/mydemo/util/ScreenUtil.kt
package com.jqk.mydemo.util

import android.content.Context

object ScreenUtil {
    /**
     * 获取屏幕密度
     * 单位：Float
     */
    fun getDensity(context: Context): Float {
        val dm = context.resources.displayMetrics

        val density = dm.density
        return density
    }

    /**
     * 获取屏幕宽度
     * 单位：Int
     */
    fun getScreenWidth(context: Context): Int {

        val dm = context.resources.displayMetrics

        val screenWidth = dm.widthPixels

        return screenWidth
    }

    /**
     * 获取屏幕高度
     * 单位：Int
     */
    fun getScreenHeight(context: Context): Int {
        val dm = context.resources.displayMetrics

        val screenHeight = dm.heightPixels

        return screenHeight
    }
}